### PR TITLE
fetches, adds, deletes and updates pokemons

### PR DESCRIPTION
**Pull request by albertoochoa**
problem: We need to create endpoints  so the IOS team can work with.
Solution: Gets info from the pokemon api and save it in my database. It deletes by id, adds new pokemons and edits 
evidence:

https://github.com/user-attachments/assets/9b69ed8a-4625-401c-8903-72e36838bb6c

### CODE REVIEW

File: handlers/addPokemonRoute.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const PokemonData = require('../models/pokemons');
+
+const addPokemon = {
+    method: 'POST',
+    handler: async function (request, h) {
+        const newPokemon = request.payload; 
+
+        if (!newPokemon.id) {
```
Comment (JuanPaHdezG) [inline handlers/addPokemonRoute.js:8]: This validation has a wrong format. It should be (pokemon.id == null)

File: handlers/addPokemonRoute.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            return h.response({ message: 'A Pokémon ID is required!' }).code(400);
+        }
+
+        try {
+            const existingPokemon = await PokemonData.findOne({ id: newPokemon.id });
+
+            if (existingPokemon) {
```
Comment (JuanPaHdezG) [inline handlers/addPokemonRoute.js:15]: Same, validation format

File: handlers/deletePokemonbyIdRoute.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const PokemonData = require('../models/pokemons');
+
+const deletePokemonById = {
+    method: 'DELETE',
+    handler: async function (request, h) {
+        const { id } = request.query;
+
+        if (!id) {
```
Comment (JuanPaHdezG) [inline handlers/deletePokemonbyIdRoute.js:8]: validation format

File: handlers/editPokemonRoute.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const PokemonData = require('../models/pokemons');
+
+const editPokemon = {
+    method: 'PUT',
+    handler: async function (request, h) {
+        const { id } = request.query;
+
+        if (!id) {
```
Comment (JuanPaHdezG) [inline handlers/editPokemonRoute.js:8]: validation in the if

File: handlers/editPokemonRoute.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            return h.response({ message: 'Please provide a valid Pokémon ID!' }).code(400);
+        }
+
+        const updates = request.payload; 
+
+        try {n
```
Comment (JuanPaHdezG) [inline handlers/editPokemonRoute.js:14]: extra character

File: handlers/getAllPokemonsRoute.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
+const PokemonData = require('../models/pokemons');
+
+const getPokemon = {
+    method: 'GET',
+    handler: async function (request, h) {
+        const { id } = request.query;
+
+        if (!id) {
```
Comment (JuanPaHdezG) [inline handlers/getAllPokemonsRoute.js:8]: validation format

File: handlers/getAllPokemonsRoute.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            return h.response({ message: 'Please provide a valid Pokémon ID!' }).code(400);
+        }
+
+        try {
+            const existingPokemon = await PokemonData.findOne({ id: parseInt(id, 10) });
+
+            if (existingPokemon) {
```
Comment (JuanPaHdezG) [inline handlers/getAllPokemonsRoute.js:15]: validation format

File: handlers/getAllPokemonsRoute.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                return h.response({
+                    message: `Fetched Pokémon ${existingPokemon.name} from the database.`,
+                    data: existingPokemon,
+                }).code(200);
+            }
+
+            const response = await fetch(`https://pokeapi.co/api/v2/pokemon/${id}/`);
+
+            if (!response.ok) {
```
Comment (JuanPaHdezG) [inline handlers/getAllPokemonsRoute.js:24]: validation format

File: handlers/getAllPokemonsRoute.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                { $set: pokemonData },
+                { upsert: true }
+            );
+
+            return h.response({
+                message: `Fetched and saved Pokémon ${pokemonData.name} from the API.`,
+                data: pokemonData,
+            }).code(200);
+        } catch (error) {
+            console.error(error);
```
Comment (JuanPaHdezG) [inline handlers/getAllPokemonsRoute.js:54]: erase this console.log

File: handlers/getAllRoute.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+
+const saveAllPokemons = {
+    method: 'GET',
+    path: '/pokemon/fetchAll',
+    handler: async function (request, h) {
+        try {
+            const savePokemon = async (id) => {
+                const response = await fetch(`https://pokeapi.co/api/v2/pokemon/${id}/`);
+                if (!response.ok) {
```
Comment (JuanPaHdezG) [inline handlers/getAllRoute.js:12]: validation format

File: handlers/getAllRoute.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+                    console.error(`Failed to fetch Pokémon with ID ${id}`);
```
Comment (JuanPaHdezG) [inline handlers/getAllRoute.js:13]: use response in stead of console.log

File: handlers/addPokemonRoute.js (hunk 1 above)
Comment (albertoochoa) [reply handlers/addPokemonRoute.js:8]: We are using joi to solve validations

**Pull request merged**